scylla/src/transport/topology.rs

```diff
@@ -902,6 +903,91 @@ async fn query_user_defined_types(
     Ok(udts)
 }
 
+fn topo_sort_udts(udts: &mut Vec<UdtRowWithParsedFieldTypes>) -> Result<(), QueryError> {
+    fn do_with_referenced_udts(what: &mut impl FnMut(&str), pre_cql_type: &PreCqlType) {
+        match pre_cql_type {
+            PreCqlType::Native(_) => (),
+            PreCqlType::Collection { type_, .. } => match type_ {
+                PreCollectionType::List(t) | PreCollectionType::Set(t) => {
+                    do_with_referenced_udts(what, t)
+                }
+                PreCollectionType::Map(t1, t2) => {
+                    do_with_referenced_udts(what, t1);
+                    do_with_referenced_udts(what, t2);
+                }
+            },
+            PreCqlType::Tuple(types) => types
+                .iter()
+                .for_each(|type_| do_with_referenced_udts(what, type_)),
+            PreCqlType::UserDefinedType { name, .. } => what(name),
+        }
+    }
+
+    // Build an indegree map: for each node in the graph, how many directly depending types it has.
+    let mut indeg = udts
+        .drain(..)
+        .map(|def| {
+            (
+                (def.keyspace_name.clone(), def.type_name.clone()),
+                (def, Cell::new(0u32)),
+            )
+        })
+        .collect::<HashMap<_, _>>();
+
+    // For each node in the graph...
+    for (def, _) in indeg.values() {
+        let mut increment_referred_udts = |type_name: &str| {
+            let deg_cell = indeg
+                .get(&(def.keyspace_name.clone(), type_name.to_string()))
+                .map(|(_, count)| count)
+                .unwrap();
```

Comment:
Hmm, what if the metadata is inconsistent and some UDT definitions are missing? This will probably crash the application.

If the dependent UDT is not present in the metadata, then we should just ignore it and not increment anything.

Please add a test which verifies that `topo_sort_udts` correctly handles inconsistent metadata.